### PR TITLE
Do not set COMPILER_HOST env variables

### PR DESCRIPTION
**Pull request by smuzaffar**
Move the `COMPILERHOST` definition in `rocm` tool ( where it is needed)

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

File: scram-tools.file/tools/rocm/env.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash -e
+export COMPILER_HOST=$(gcc -dumpmachine)
+export COMPILER_PATH=${GCC_ROOT}
```
Comment (fwyzard) [inline scram-tools.file/tools/rocm/env.sh]: Does this set the path to what it is at runtime (after the installation and `cmsrel ...`) or what it was when the external itself was built ?

Comment (smuzaffar) [reply scram-tools.file/tools/rocm/env.sh]: yes, the path will be relocated at install time (just it it is done for other toolfiles)

Comment (smuzaffar) [reply scram-tools.file/tools/rocm/env.sh]: this file is only used at build time and the `@COMPILER_PATH@` in [scram-tools.file/tools/rocm/rocm.xml](https://github.com/cms-sw/cmsdist/pull/8087/files#diff-b43b9ccc937203603b2df4858063a3278b4c872b00e2c30a3c3f2e899e068c88) is replaced with the value of $COMPILER_PATH. At install time , we relocate all toolfiles

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
+externals

**Pull request merged**